earth2studio/models/px/dlesym.py

```diff
@@ -0,0 +1,1077 @@
+        hpx_lon: np.ndarray,
+        nside: int,
+        center: np.ndarray,
+        scale: np.ndarray,
+        atmos_constants: np.ndarray,
+        ocean_constants: np.ndarray,
+        atmos_input_times: np.ndarray,
+        ocean_input_times: np.ndarray,
+        atmos_output_times: np.ndarray,
+        ocean_output_times: np.ndarray,
+        atmos_variables: list[str],
+        ocean_variables: list[str],
+        atmos_coupling_variables: list[str],
+        ocean_coupling_variables: list[str],
+    ):
+
+        super().__init__()
+        self.atmos_model = atmos_model.eval()
+        self.ocean_model = ocean_model.eval()
+
+        self.register_buffer("center", torch.from_numpy(center).to(dtype=torch.float32))
+        self.register_buffer("scale", torch.from_numpy(scale).to(dtype=torch.float32))
+        self.register_buffer(
+            "atmos_constants", torch.from_numpy(atmos_constants).to(dtype=torch.float32)
+        )
+        self.register_buffer(
+            "ocean_constants", torch.from_numpy(ocean_constants).to(dtype=torch.float32)
+        )
+
+        self.hpx_lat = hpx_lat
+        self.hpx_lon = hpx_lon
+        self.nside = nside
+        self.atmos_variables = atmos_variables
+        self.ocean_variables = ocean_variables
+        self.atmos_coupling_variables = atmos_coupling_variables
+        self.ocean_coupling_variables = ocean_coupling_variables
+
+        # Validate the input and output times
+        for name, times in zip(
+            [
+                "atmos_input_times",
+                "ocean_input_times",
+                "atmos_output_times",
+                "ocean_output_times",
+            ],
+            [
+                atmos_input_times,
+                ocean_input_times,
+                atmos_output_times,
+                ocean_output_times,
+            ],
+        ):
+            if not np.issubdtype(times.dtype, np.timedelta64):
+                raise ValueError(
+                    f"Input and output times must be of type 'timedelta64', got {times.dtype} for {name}"
+                )
+        if ocean_input_times[0] > atmos_input_times[0]:
+            raise ValueError(
+                f"Ocean input time must be equal to or before atmos input time, got {ocean_input_times[0]} and {atmos_input_times[0]}"
+            )
+        if len(atmos_input_times) < len(ocean_input_times):
+            raise ValueError(
+                f"Atmos input times must be equal to or longer than ocean input times, got {len(atmos_input_times)} and {len(ocean_input_times)}"
+            )
+        if len(atmos_output_times) < len(ocean_output_times):
+            raise ValueError(
+                f"Atmos output times must be equal to or longer than ocean output times, got {len(atmos_output_times)} and {len(ocean_output_times)}"
+            )
+
+        self.atmos_input_times = atmos_input_times
+        self.ocean_input_times = ocean_input_times
+        self.full_input_times = np.arange(
```

Comment:
I'm having trouble following what the full_input_times is, its a index list?
I suppose I was kind of expecting a np.concat(atmost_times, ocean_times)

Reply:
It's more of a merge of the atmos_times and ocean_times rather than a concat -- we need data as far back as `-48h` for the ocean model, whereas we need data on `6h` time resolution for the atmos model; ultimately this all has to go into one `(Tensor, CoordSystem)` pair as input to the model. An easy way to satisfy this is to get the full 48h of data at 6h resolution and then pull out what we need. Now that I think of it though, a simple concat could work too, as long as the `prepare_input_data` method correctly pulls out the desired ocean and atmos timesteps. Might be more efficient, but a bit less intuitive for preparing input data tensors

Reply:
Ah I see, okay I'm following.